test/gurka/test_permissible_dimensions.cc

```diff
@@ -0,0 +1,112 @@
+#include "gurka.h"
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+namespace {
+const std::vector<std::string> kSupportedCostingTypes = {"auto", "truck", "bus", "taxi", "hov"};
+const std::vector<std::string> kUnsupportedCostingTypes = {"bicycle", "pedestrian", "motorcycle",
+                                                           "motor_scooter"};
+} // namespace
+
+TEST(Standalone, Maxwidth) {
+  const std::string ascii_map = R"(
+      A----B----C----D----E
+           |         |    |
+           |         |    |
+           I----H----G----F
+  )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "residential"}}},
+      {"BC", {{"highway", "residential"}, {"maxwidth", "2.0"}}},
+      {"CD", {{"highway", "residential"}, {"maxwidth", "2.0m"}}},
+      {"DE", {{"highway", "residential"}}},
+      {"EF", {{"highway", "residential"}}},
+      {"FG", {{"highway", "residential"}}},
+      {"GH", {{"highway", "residential"}}},
+      {"HI", {{"highway", "residential"}}},
+      {"DG", {{"highway", "residential"}, {"maxwidth", "1.9"}}},
+      {"BI", {{"highway", "residential"}, {"maxwidth", "1.9m"}}},
+
+  };
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 100);
+  const auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_maxwidth");
+
+  for (const auto& costing : kSupportedCostingTypes) {
+    const auto result = gurka::do_action(valhalla::Options::route, map, {"A", "I"}, costing,
+                                         {{"/costing_options/" + costing + "/width", "2.0"}});
+    ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "EF", "FG", "GH", "HI"});
+  }
+
+  for (const auto& costing : kSupportedCostingTypes) {
+    const auto result = gurka::do_action(valhalla::Options::route, map, {"A", "I"}, costing,
+                                         {{"/costing_options/" + costing + "/width", "1.8"}});
```

Comment:
@maksimandrianov can we get a test with 1.9.

Reply:
Yes, I have added